packages/neuron-wallet/src/services/node.ts

```diff
@@ -219,38 +234,35 @@ class NodeService {
     }
   }
 
-  private async verifyNodeVersion() {
+  private async verifyCkbNodeVersion() {
     const network = NetworksService.getInstance().getCurrent()
     const localNodeInfo = await new RpcService(network.remote).localNodeInfo()
     const internalNodeVersion = this.getInternalNodeVersion()
-    const [internalMajor, internalMinor] = internalNodeVersion?.split('.') ?? []
-    const [externalMajor, externalMinor] = localNodeInfo.version?.split('.') ?? []
+    const [internalMajor, internalMinor] = internalNodeVersion?.split('.')?.map(v => +v) ?? []
+    const [externalMajor, externalMinor] = localNodeInfo.version?.split('.')?.map(v => +v) ?? []
 
-    if (internalMajor !== externalMajor || (externalMajor === '0' && internalMinor !== externalMinor)) {
-      dialog.showMessageBox({
-        type: 'warning',
-        message: t('messageBox.node-version-different.message', { version: internalNodeVersion }),
-      })
+    if (internalMajor === externalMajor && internalMinor === externalMinor) return VerifyCkbVersionResult.Same
+    if (internalMajor < externalMajor || (internalMajor === externalMajor && internalMinor < externalMinor)) {
+      return VerifyCkbVersionResult.ShouldUpdate
+    }
+    if (supportCkbVersions.every(v => !localNodeInfo.version.startsWith(v))) {
+      return VerifyCkbVersionResult.NotCompatibility
     }
```

Comment:
The comparison should be an independent function for testing.

Reply:
Do you mean to add test cases for the function? https://github.com/yanguoyu/neuron/blob/feat-ckb-check/packages/neuron-wallet/tests/services/node.test.ts#L455-L504 is the test case.

Reply:
IMO, `verifyCkbNodeVersion` is small, should I split this function into two functions `getCKbVersion` and `veifyCkbVersion`?

Reply:
I would define a utility function `getCompatibility(neuronVersion, nodeVersion)` to return the compatibility and tested by a set of test cases.

Then call it in `verifyExternalCkbNode()` with `getExternalCkbNodeVersion()` and `getNeuronVersion()`

```js
const nodeVersion = getExternalCkbNodeVersion()
const neuronVersion = getNeuronVersion()
return getCompatibility(neuronVersion, nodeVersion)
```

By doing this, only `getExternalCkbNodeVersion` should be tested with mock. Or not, because the external ckb node version can be fetched by RPC